motion_control.c

```diff
@@ -338,8 +339,20 @@ void mc_reset()
     // the steppers enabled by avoiding the go_idle call altogether, unless the motion state is
     // violated, by which, all bets are off.
     if (sys.state & (STATE_CYCLE | STATE_HOLD | STATE_HOMING)) {
+      sys.alarm |= ALARM_ABORT_CYCLE;  //killed while in motion
       SYS_EXEC |= EXEC_ALARM; // Flag main program to execute alarm state.
       st_go_idle(); // Force kill steppers. Position has likely been lost.
     }
   }
 }
+
+
+// Method to stop motion in progress and clear the queues
+void mc_abort() { 
+  SYS_EXEC |= EXEC_FEED_HOLD;  //Stop motion;
+  while (sys.state & (STATE_CYCLE)) protocol_execute_runtime(); //spin untill stopped
```

Comment:
for clarity, you should at the very least use some {}'s.  I'd prefer the function on it's own line so the indent will should what's happening.